### PR TITLE
Improve kubeadm upgrade apply  error logging style

### PR DESCRIPTION
**Pull request by wackxu**
**What this PR does / why we need it**:

Improve kubeadm upgrade apply  error logging style

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # https://github.com/kubernetes/kubeadm/issues/549

**Special notes for your reviewer**:

/assign @luxas 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/cmd/upgrade/apply.go [hunk 1]
```diff
@@ -202,16 +202,16 @@ func EnforceVersionPolicies(flags *applyFlags, versionGetter upgrade.VersionGett
 	if versionSkewErrs != nil {
 
 		if len(versionSkewErrs.Mandatory) > 0 {
-			return fmt.Errorf("The --version argument is invalid due to these fatal errors: %v", versionSkewErrs.Mandatory)
+			return fmt.Errorf("The --version argument is invalid due to these fatal errors:\n\n%v\nPlease fix the misalignments highlighted above and try upgrading again", kubeadmutil.FormatErrMsg(versionSkewErrs.Mandatory))
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/upgrade/apply.go]: You changed `The` to `the` in the similar sentence below. Please choose whether you think we should start with a capital letter or not, but be consistent in usage.

File: cmd/kubeadm/app/util/error.go [hunk 2]
```diff
@@ -75,3 +75,11 @@ func checkErr(prefix string, err error, handleErr func(string, int)) {
 		handleErr(err.Error(), DefaultErrorExitCode)
 	}
 }
+
+func FormatErrMsg(errs []error) string {
```
Comment (luxas) [inline cmd/kubeadm/app/util/error.go]: this needs a godoc comment `// FormatErrMsg returns a human-readable string describing the slice of errors passed to the function`

Comment (luxas) [reply cmd/kubeadm/app/util/error.go]: Please also add an unit test here

**Review by luxas — COMMENTED** [review]
One small nit and a request for unit tests

File: cmd/kubeadm/app/cmd/upgrade/apply.go (hunk 1 above)
Comment (wackxu) [reply cmd/kubeadm/app/cmd/upgrade/apply.go]: according to https://github.com/golang/go/wiki/CodeReviewComments#error-strings and follow up #54280.  We should not start with a capital letter. But it is just a weak error and there are so many cases, I can commit another pr to fix it.

**Comment by wackxu** [comment]
@luxas Done, PTAL

**Review by luxas — APPROVED** [review]
Thanks @wackxu for the great contribution!
/lgtm
/approve

**Pull request merged**